### ISSUE TITLE
Doubts about the evaluation

### ISSUE THREAD

**Issue by chinmay5** [NILOIDE/Implicit_segmentation#1]
Thank you so much for releasing the code. However, I had a doubt with the way the model checkpoints are saved.

`ckpt_latest_saver = ModelCheckpoint(save_top_k=1, dirpath=root_dir / LATEST_CHECKPOINT_DIR,
                                        monitor="step", mode="max")`

Is there a reason not to save checkpoints based on the validation loss and just save the latest checkpoint? As far as I can understand, the `monitor=step` implies there is no specific criterion for saving the model checkpoints, just its epoch number. Please correct me if I am wrong.

Best,
Chinmay

**Comment by chinmay5** [comment]
Sorry, I figured it out. You are making sure to save the model based on the best validation loss but doing it explicitly.

**Comment by NILOIDE** [comment]
I could be mistaken, but there are some issues with having multiple instances of ModelCheckpoint in Pytorch lightning that caused us to not want to deal with it.
